src/packageurl/__init__.py

```diff
@@ -546,4 +541,4 @@ def from_string(cls, purl: str) -> "PackageURL":
             encode=False,
         )
 
-        return PackageURL(type, namespace, name, version, qualifiers, subpath)
+        return cls(type, namespace, name, version, qualifiers, subpath)  # type: ignore[arg-type]
```

Comment:
you can easily fix this, then I don't need to create a separate PR 😅 

Just create your own `AnyStr` type alias and remove the one coming from the `typing` module.

Add
```python
AnyStr = Union[str, bytes]
```
directly under `basestring` at the top and remove the `from typing import AnyStr`. `AnyStr` was deprecated in Python 3.13 anyway.

Reply:
Good call! One small alteration: since it's only used for type hints, I put it into the `TYPE_CHECKING` block. Tests still pass locally. Are you good with that?

Reply:
sure, also sounds good